lib/rbac/filterer.rb

```diff
@@ -514,17 +514,19 @@ def scope_for_user_role_group(klass, scope, miq_group, user, managed_filters)
       if user_or_group.try!(:self_service?) && MiqUserRole != klass
         scope.where(:id => klass == User ? user.id : miq_group.id)
       else
-        if user_or_group.miq_user_role_name == 'EvmRole-tenant_administrator'
-          scope = scope.with_roles_excluding(%w(EvmRole-super_administrator EvmRole-administrator))
+        # hide creating admin group / roles from tenant administrators
+        if !user_or_group.miq_user_role&.admin_user?
+          scope = scope.with_roles_excluding([MiqProductFeature::SUPER_ADMIN_FEATURE, MiqProductFeature::ADMIN_FEATURE])
         end
 
-        if MiqUserRole != klass
+        if MiqUserRole == klass
+          scope
```

Comment:
What is the rational behind adding this conditional in and pushing `scope_by_ids` inside the else?  Just don't see the context/rational from the diff.

Reply:
Not sure you need to go that far, was just curious what the rational was.